File: 349.intersection-of-two-arrays.0.go
/*
 * @lc app=leetcode id=349 lang=golang
 *
 * [349] Intersection of Two Arrays
 */
func intersection(nums1 []int, nums2 []int) (ret []int) {
    s1 := map[int]struct{}{}
    s2 := map[int]struct{}{}

    for _, n := range nums1 {
        s1[n] = struct{}{}
    }
    for _, n := range nums2 {
        s2[n] = struct{}{}
    }

    for n, _ := range s1 {
        if _, ok := s2[n]; ok {
            ret = append(ret, n)
        }
    }
    return
}

File: 349.intersection-of-two-arrays.0.kt
/*
 * @lc app=leetcode id=349 lang=kotlin
 *
 * [349] Intersection of Two Arrays
 */
class Solution_intersection_0 {
    fun intersection(nums1: IntArray, nums2: IntArray): IntArray {
        val s1 = nums1.toSet()
        val s2 = nums2.toSet()
        return (s1 intersect s2).toIntArray()
    }
}
